api/nbaData.js

```diff
@@ -4,36 +4,79 @@ const fetch = require('node-fetch');
  */
 getPlayers = function() {
     return fetch('http://data.nba.net/10s/prod/v1/2018/players.json')
-        .then(res => {
-            return res.json();
-        })
-        .then(response => {
-            nbaData = response.league.standard
-            return nbaData;
-        })
-        .catch(err => {
-            console.log(`${err} Promise error`);
-        })
-    }
+    .then(res => {
+        return res.json();
+    })
+    .then(response => {
+        nbaData = response.league.standard
+        return nbaData;
+    })
+    .catch(err => {
+        console.log(`${err} Promise error`);
+    })
+}
+
+
+/// Might want to refactor nbaData to return data this way
+getPlayersTest = async function(){
+    const url = 'http://data.nba.net/10s/prod/v1/2018/players.json';
+    const players = await fetch(url)
+    .then(resp => resp.json())
+    .then(resp => (resp));
+    return players;
+}
 
 /**
  * @param {number} teamId Unique key for nba team
  * @return {Promise<pending>} All given team games
  */
 getTeamSchedule = function(teamId) {
     return fetch(`http://data.nba.net/10s/prod/v1/2018/teams/${teamId}/schedule.json`)
-        .then(res => {
-            return res.json();
-        })
-        .then(response => {
-            const data = response.league.standard
-            return data;
-
-        })
-        .catch(err => {
-            console.log(`${err} That teamId doesn't exist`);
-        })
-    }
+    .then(res => {
+        return res.json();
+    })
+    .then(response => {
+        const data = response.league.standard
+        return data;
+
+    })
+    .catch(err => {
+        console.log(`${err} That teamId doesn't exist`);
+    })
+}
+
+/**
+ * @param {string} teamId Unique key for nba team
+ * @return {number} index of last game played by TeamId
+ */
+getLastPlayedIndex = function(teamId){
+    return fetch(`http://data.nba.net/10s/prod/v1/2018/teams/${teamId}/schedule.json`)
+    .then(res => {
+        return res.json();
+    })
+    .then(response => {
+        const data = response.league.lastStandardGamePlayedIndex;
+        return data;
+
+    })
+    .catch(err => {
+        console.log(`${err} That teamId doesn't exist`);
+    })
+}
+
+
+// refactored getTeamSchedule
+getTeamScheduleTest = async function(teamId){

```

Comment:
If this works lets change all endpoint methods to returns JSON and not promise<pending>